src/ucp/tag/rndv.c

```diff
@@ -1535,7 +1535,7 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_rndv_atp_handler,
     ucp_reply_hdr_t *rep_hdr = data;
     ucp_request_t *req       = (ucp_request_t*) rep_hdr->reqptr;
 
-    if (req->recv.frag.rreq) {
+    if (req->flags & UCP_REQUEST_FLAG_RNDV_FRAG) {
```

Comment:
so we cannot use req->recv.frag.rreq, because it's in a union?
maybe add ucs_assert(req->recv.frag.rreq != NULL)?